src/unix/process.c

```diff
@@ -454,9 +455,9 @@ int uv_spawn(uv_loop_t* loop,
   if (r == 0)
     ; /* okay, EOF */
   else if (r == sizeof(exec_errorno))
-    ; /* okay, read errorno */
+    waitpid(pid, &status, 0); /* okay, read errorno */
   else if (r == -1 && errno == EPIPE)
-    ; /* okay, got EPIPE */
+    waitpid(pid, &status, 0); /* okay, got EPIPE */
```

Comment:
You have to handle EINTR here and it should probably abort on error, i.e.:

```c
do
  err = waitpid(pid, &status, 0);
while (err == -1 && errno == EINTR);
assert(err == pid);
```

It would be nice to have a regression test.